python/tvm/relax/op/manipulate.py

```diff
@@ -219,3 +216,60 @@ def broadcast_to(data: Expr, shape: Union[Tuple[PrimExprLike], Expr]) -> Expr:
     if isinstance(shape, (tuple, list)):
         shape = ShapeExpr(shape)
     return _ffi_api.broadcast_to(data, shape)  # type: ignore
+
+
+def collapse_sum_like(data: Expr, collapse_target: Expr) -> Expr:
+    """Return a summation of data to the shape of collapse_target.
+
+    For details, please see relax.op.collapse_sum_to.
+
+    Parameters
+    ----------
+    data : relax.Expr
+        The input tensor.
+
+    collapse_target : relax.Expr
+        The tensor whose shape is the shape to collapse to.
+
+    Returns
+    -------
+    result : relax.Expr
+        The result tensor after summation.
+    """
+    return _ffi_api.collapse_sum_like(data, collapse_target)
+
+
+def collapse_sum_to(
+    data: Expr, shape: Union[List[PrimExprLike], Tuple[PrimExprLike], Expr]
+) -> Expr:
+    """Return a summation of data to the given shape.
+
+    collapse_sum_to is intended as the backward operator of tvm.relax.op.broadcast_to and
+    other broadcast operators in the automatic differentiation process.
+
+    We expect that data is the result of broadcasting some tensor of the given shape in some
+    broadcast operation. Thus the given shape and data.shape must follow broadcast rules.
+
+    During computation, the axes of data.shape and shape are checked from right to left. For every
+    axis, if it either:
+    - exist in data but not in collapse_target, or
+    - is larger than 1 in data and equals to 1 in collapse_target,
+
+    data will be summed over this axis.
+
+    Parameters
+    ----------
+    data : relax.Expr
+        The input tensor.
+
+    shape : Union[List[PrimExprLike], Tuple[PrimExprLike], relax.Expr]
```

Comment:
Follow the convention of broadcast_to
```suggestion
    shape : Union[Tuple[PrimExprLike], relax.Expr]
```

Actually it is a little weird because the annotation of broadcast_to is `Union[Tuple[PrimExprLike], Expr]` , but in code it writes
```
    if isinstance(shape, (tuple, list)):
        shape = ShapeExpr(shape)
```
which has one more choice for list. This issue also exists in create ops.
@MasterJH5574

Reply:
This is intended to avoid overlong type annotation. I admit this could be a bit informal but it should be fine I think.